gbpservice/nfp/core/controller.py

```diff
@@ -344,245 +427,122 @@ def get_stashed_events(self):
-            (modules_dir))
-        files = []
-
-    for fname in files:
-        if fname.endswith(".py") and fname != '__init__.py':
-            try:
-                module = __import__(oslo_config.CONF.modules_dir,
-                                    globals(), locals(), [fname[:-3]], -1)
-                modules += [eval('module.%s' % (fname[:-3]))]
-            except Exception as exc:
-                LOG.error("NFP module %s import failed." % (fname))
-
-    sys.path = syspath
-    return modules
-
-
-def common_init():
-    oslo_config.CONF.register_opts(nfp_config.OPTS)
-    oslo_config.CONF.register_opts(
-        nfp_config.es_openstack_opts, "keystone_authtoken")
-
-    # Since other imports are registering the logging configuration
-    # parameters, these are overridden to make sure that the core
-    # configuration parameters are effective.
-    oslo_config.CONF.set_override('use_syslog', 'True')
-    oslo_config.CONF.set_override('syslog_log_facility', 'local1')
-
-    # n_config.register_interface_driver_opts_helper(oslo_config.CONF)
-    # n_config.register_agent_state_opts_helper(oslo_config.CONF)
-    # n_config.register_root_helper(oslo_config.CONF)
-
-    n_common_config.init(sys.argv[1:])
-    n_config.setup_logging()
-
+        base_module = __import__(conf.nfp_modules_path,
+                                 globals(), locals(), ['modules'], -1)
+        modules_dir = base_module.__path__[0]
+        try:
+            files = os.listdir(modules_dir)
+            for pyfile in set([f for f in files if f.endswith(".py")]):
+                try:
+                    pymodule = __import__(conf.nfp_modules_path,
+                                          globals(), locals(),
+                                          [pyfile[:-3]], -1)
+                    pymodule = eval('pymodule.%s' % (pyfile[:-3]))
+                    try:
+                        pymodule.nfp_module_init(controller, conf)
+                        pymodules += [pymodule]
+                        LOG.debug("(module - %s) - Initialized" %
+                                  (identify(pymodule)))
+                    except AttributeError as e:
+                        import sys
+                        import traceback
+                        exc_type, exc_value, exc_traceback = sys.exc_info()
+                        print traceback.format_exception(exc_type, exc_value,
+                                                         exc_traceback)
+                        LOG.error("(module - %s) - "
+                                  "does not implement"
+                                  "nfp_module_init()" % (identify(pymodule)))
+                except ImportError:
+                    LOG.error(
+                        "Failed to import module %s" % (pyfile))
+        except OSError:
+            LOG.error(
+                "Failed to read files from %s" % (modules_dir))
+    except ImportError:
+        LOG.error(
```

Comment:
Why to proceed on any exception, or atleast on Import Error ?
We should not proceed further.